src/utils/filterFunction.ts

```diff
@@ -0,0 +1,95 @@
+import { SearchParametersType, UnprocessedResultsFromCRM } from '../types'
+
+export default function sortResultsFunction (rawUnsortedPropertyResults: string, searchParameters: SearchParametersType[]): UnprocessedResultsFromCRM[] {
+    const unsortedPropertyResults = JSON.parse(rawUnsortedPropertyResults)
+
+    const maxNumNeigbours = searchParameters[0].neighboursSearchMaxRecords
+    const maxResultsForPropertyTypes = searchParameters[0].propertyTypesMaxResults
+    const maxResultsForPropertyGroups = searchParameters[0].propertyGroupsMaxResults
+    const desiredPropertyTypes = searchParameters[0].propertyTypes
+    const desiredPropertyGroups = searchParameters[0].propertyGroups
+    const managed = searchParameters[0].managed[0]
+
+    let totalNeighboursAdded = 0
+    let totalPropertyTypeAdded = 0
+    let totalPropertyGroupAdded = 0
+
+    const propertyDistances: string[] = Object.keys(unsortedPropertyResults[0])
+
+    const sortedPropertyDistances = propertyDistances.sort((propertyDistance1: any, propertyDistance2: any) => {
+        return propertyDistance1.split('dist')[1] - propertyDistance2.split('dist')[1]
+    })
+
+    const propertyMap: any = {}
+    const matchedProperties: any = []
+    sortedPropertyDistances.forEach((propertyDistance: string) => {
```

Comment:
this inner function is very complicated (I know it was originally in deluge because I don't have the luxury of calling other functions in deluge). I suggest breaking it up into several smaller functions:

```
const matchTallies = {
  propertyType: 0,
  propertyGroup: 0,
  neighbour: 0,
};

sortedPropertyDistances.forEach((propertyDistance: string) => {
  const canAddAnotherProperty =
    matchTallies.neighbour < maxNumNeigbours ||
    matchTallies.propertyType < maxResultsForPropertyTypes ||
    matchTallies.propertyGroup < maxResultsForPropertyGroups;

  if (canAddAnotherProperty) {
    const { propertyMatches, typeOfMatch } = propertyMeetsFilters(
      property,
      searchParameters
    );
    if (!propertyMatches) {
      return;
    }

    matchTallies[typeOfMatch] += 1;
    const ownerData = getOwnerData(property);

    matchedProperties.push(property);
    propertyMap[propertyAddress] = "found";
  }
});
```